sympy/solvers/solveset.py

```diff
@@ -3694,4 +3694,17 @@ def nonlinsolve(system, *symbols):
         # Use `substitution` method for the system
         result = substitution(
             polys_expr + nonpolys, symbols, exclude=denominators)
-        return result
+    valid_solns = []
+    if isinstance(result, FiniteSet):
```

Comment:
```suggestion
    if isinstance(result, FiniteSet):
    valid_solns = []
```

Reply:
Seems like a good suggestion. I've committed the change.